src/config.c

```diff
@@ -33,12 +113,462 @@ const char *ChunkTypeToString(int options) {
+
+        if (!value) {
+            return NULL;
+        }
+
+        if (tempStringStringToJustReturnInTheGetter) {
+            RedisModule_FreeString(rts_staticCtx, tempStringStringToJustReturnInTheGetter);
+        }
+
+        tempStringStringToJustReturnInTheGetter =
+            RedisModule_CreateString(rts_staticCtx, value, strlen(value));
+
+        return tempStringStringToJustReturnInTheGetter;
+    } else if (!strcasecmp("ts-encoding", name)) {
+        const char *value = ChunkTypeToString(TSGlobalConfig.options);
+
+        if (!value) {
+            return NULL;
+        }
+
+        if (tempStringStringToJustReturnInTheGetter) {
+            RedisModule_FreeString(rts_staticCtx, tempStringStringToJustReturnInTheGetter);
+        }
+
+        tempStringStringToJustReturnInTheGetter =
+            RedisModule_CreateString(rts_staticCtx, value, strlen(value));
+
+        return tempStringStringToJustReturnInTheGetter;
+    } else if (!strcasecmp("ts-ignore-max-val-diff", name)) {
+        if (tempStringStringToJustReturnInTheGetter) {
+            RedisModule_FreeString(rts_staticCtx, tempStringStringToJustReturnInTheGetter);
+        }
+
+        tempStringStringToJustReturnInTheGetter =
+            RedisModule_CreateStringPrintf(rts_staticCtx, "%lf", TSGlobalConfig.ignoreMaxValDiff);
+
+        return tempStringStringToJustReturnInTheGetter;
+    }
+
+    return NULL;
+}
+
+/*
+ * Parses a string value and validate it is a valid double in the range
+ * [min, max].
+ *
+ * If the value is invalid, set `err` to an error message.
+ * Otherwise, set `value` to the parsed value and return it without
+ * setting the error string.
+ *
+ * The success of the operation can be determined by checking if `err`
+ * is NULL.
+ */
+static double stringToDouble(const char *name,
+                             RedisModuleString *string,
+                             const double min,
+                             const double max,
+                             RedisModuleString **err) {
+    double value = 0.0;
+    long long longValue = 0;
+
+    if (RedisModule_StringToLongLong(string, &longValue) == REDISMODULE_OK) {
+        value = (double)longValue;
+    } else if (RedisModule_StringToDouble(string, &value) != REDISMODULE_OK) {
+        *err = RedisModule_CreateStringPrintf(rts_staticCtx, "Invalid value for `%s`", name);
+    }
+
+    if (value < min || value > max) {
+        *err = RedisModule_CreateStringPrintf(
+            rts_staticCtx,
+            "Invalid value for `%s`. Value must be in the range [%f .. %f]",
+            name,
+            min,
+            max);
+    }
+
+    return value;
+}
+
+static int setModernStringConfigValue(const char *name,
+                                      RedisModuleString *value,
+                                      void *data,
+                                      RedisModuleString **err) {
+    if (!strcasecmp("ts-compaction-policy", name)) {
```

Comment:
Why not have different functions that each handle their own specific config variable? I would understand if there were a load of general work to be done with a little specific variation per config, but this is the opposite. Each config is being handled wholly separately and should be handled by its own dedicated function.